ivi-layermanagement-examples/layer-add-surfaces/src/layer-add-surfaces.c

```diff
@@ -96,6 +96,35 @@ static void callbackFunction(ilmObjectType object, t_ilm_uint id, t_ilm_bool cre
     }
 }
 
+static void shutdownCallbackFunction(t_ilm_shutdown_error_type error_type,
+                                     int errornum,
+                                     void *user_data)
+{
+    (void) user_data;
+
+    switch (error_type) {
+        case ILM_ERROR_WAYLAND:
+        {
+            printf("layer-add-surfaces: ilm shutdown due to wayland error: %s\n",
+                   strerror(errornum));
+            break;
+        }
+        case ILM_ERROR_POLL:
+        {
+            printf("layer-add-surfaces: ilm shutdown due to poll error: %s\n",
+                   strerror(errornum));
+            break;
+        }
+        default:
+        {
+            printf("layer-add-surfaces: ilm shutdown due to unhandled error: %s\n",
+                   strerror(errornum));
```

Comment:
shutdown could also happen because of the restarting of the compositor, clear we are not expecting this in normal operation but still the message should be generic, like shutdown due unkown reason. 
and all messages should mention that we are existing, and we should exit with 1 instead of 0